packages/contracts-bedrock/scripts/DeployImplementations.s.sol

```diff
@@ -217,37 +270,136 @@ contract DeployImplementations is Script {
         deployDelayedWETHImpl(_dsi, _dso);
         deployPreimageOracleSingleton(_dsi, _dso);
         deployMipsSingleton(_dsi, _dso);
+        deployDisputeGameFactoryImpl(_dsi, _dso);
+
+        // Deploy the OP Stack Manager with the new implementations set.
+        deployOPStackManager(_dsi, _dso);
 
         _dso.checkOutput();
     }
 
     // -------- Deployment Steps --------
 
-    function deploySystemConfigImpl(DeployImplementationsInput, DeployImplementationsOutput _dso) public {
+    // --- OP Stack Manager ---
+
+    function opsmSystemConfigSetter(
+        DeployImplementationsInput,
+        DeployImplementationsOutput _dso
+    )
+        internal
+        view
+        virtual
+        returns (OPStackManager.ImplementationSetter memory)
+    {
+        return OPStackManager.ImplementationSetter({
+            name: "SystemConfig",
+            info: OPStackManager.Implementation(address(_dso.systemConfigImpl()), SystemConfig.initialize.selector)
+        });
+    }
+
+    function createOPSMContract(
+        DeployImplementationsInput _dsi,
+        DeployImplementationsOutput,
+        OPStackManager.Blueprints memory blueprints
+    )
+        internal
+        virtual
+        returns (OPStackManager opsm_)
+    {
+        SuperchainConfig superchainConfigProxy = _dsi.superchainConfigProxy();
+        ProtocolVersions protocolVersionsProxy = _dsi.protocolVersionsProxy();
+
+        vm.broadcast(msg.sender);
+        opsm_ = new OPStackManager({
```

Comment:
Should the `OPStackManager` be proxied? 

Is there a scenario when we will need to upgrade the `OPStackManager` for a chain? If so, how does that flow look? 

Let's say we wanted to update the `OPStackManager` to include a new upgrade function. How do we do this with the current approach?

Reply:
Oh, nvm I see the comment about this and how it will be proxied in the future: https://github.com/ethereum-optimism/optimism/blob/e7278178031f8a1093c25b32ac8232d07d3c3509/packages/contracts-bedrock/src/L1/OPStackManager.sol#L154

Reply:
Yep, exactly! We will move it behind a proxy in a future PR, because we do want to be able to upgrade it